c_src/quicer_connection.c

```diff
@@ -472,22 +494,47 @@ ServerConnectionCallback(HQUIC Connection,
 
   return status;
 }
+
 /*
 ** Open connection handle only
-** No ownership, No monitoring.
+** Set ownership but no monitoring.
 */
 ERL_NIF_TERM
-open_connection0(ErlNifEnv *env,
-                 __unused_parm__ int argc,
-                 __unused_parm__ const ERL_NIF_TERM argv[])
+open_connectionX(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[])
 {
   QUIC_STATUS Status = QUIC_STATUS_SUCCESS;
   ERL_NIF_TERM eHandle;
   ERL_NIF_TERM res = ERROR_TUPLE_2(ATOM_ERROR_INTERNAL_ERROR);
+  QuicerRegistrationCTX *r_ctx = NULL;
+  HQUIC registration = NULL;
+  ERL_NIF_TERM options = argv[1];
```

Comment:
Q: should we assert that `argc >= 2` before accessing this index?

Reply:
yes, the current assertion only works with TEST macro and it will cause compiler warning, unused args without TEST macro.   

currently, they are safe since number of args are defined here: https://github.com/emqx/quic/blob/main/c_src/quicer_nif.c#L1480,
but I will fix them all together when I add unit test for C code.